src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java

```diff
@@ -668,6 +674,43 @@ public void onOnline(Computer c, TaskListener listener) throws IOException, Inte
                         "Could not install root log handler on node: " + c.getName(), e);
             }
         }
+
+        @Override
+        public void onOffline(@Nonnull final Computer c, @CheckForNull OfflineCause cause) {
+            if (Boolean.getBoolean(SupportPlugin.class.getName() + ".generateOnOfflineNode")
+                    && cause != null && !cause.getClass().getName().contains("RestartCause")) {
+                Timer.get().schedule(new java.util.concurrent.Callable<Void>() {
+                    @Override
+                    public Void call() throws Exception {
+                        boolean locked = false;
+                        try {
+                            locked = lock.tryLock();
+                            if (locked) {
+                                PeriodicWork.all().get(PeriodicWorkImpl.class).doRun();
```

Comment:
IIUC this will generate a new bundle for every disconnection event, even if you get a bunch of them in sequence. Intentional? I thought we discussed coalescing these somehow?